code/game/machinery/engi_adv.dm

```diff
@@ -0,0 +1,250 @@
+// Construction "nuke"
+
+/obj/machinery/construction_nuke
+	name = "nuclear fission construction device"
+	desc = "The next level of interior redecoration."
+	icon = 'icons/obj/machines/nuke.dmi'
+	icon_state = "nuclearbomb0"
+	density = TRUE
+	use_power = 0
+
+	var/timer_set = 90
+	var/ui_style = "nanotrasen"
+	var/range = 160
+
+	var/timing = FALSE
+	var/detonation_timer
+	var/cooldown = 0
+	var/safety = TRUE
+
+	var/bomb_set = FALSE
+	var/exploding = FALSE
+	var/quiet = FALSE
+	var/payload = "plasteel"
+	var/payload_wall = /turf/closed/wall/r_wall
+	var/payload_floor = /turf/open/floor/engine
+	var/static/list/possible_payloads = list("wood","sand","ice","mining","silver","gold","bananium","abductor","desolation", "plasma","uranium","bluespace","diamond","plasteel","safety","titanium","plastitanium")
+
+/obj/machinery/construction_nuke/Initialize()
+	. = ..()
+	GLOB.poi_list += src
+
+/obj/machinery/construction_nuke/Destroy()
+	GLOB.poi_list -= src
+	return ..()
+
+/obj/machinery/construction_nuke/examine(mob/user)
+	. = ..()
+	if(timing)
+		to_chat(user, "There are [get_time_left()] seconds until detonation.")
+
+/obj/machinery/construction_nuke/update_icon()
+	. = ..()
+	if(safety)
+		icon_state = "nuclearbomb1"
+	else
+		icon_state = "nuclearbomb2"
+	if(timing)
+		if(get_time_left() <= 14)
+			icon_state = "nuclearbomb3"
+		else
+			icon_state = "nuclearbombc"
+	
+		
+
+/obj/machinery/construction_nuke/process()
+	if(timing)
+		bomb_set = TRUE
+		if(detonation_timer < world.time && !exploding)
+			explode()
+		else
+			switch(get_time_left())
+				if (30 to 3600)
+					playsound(src, 'sound/items/timer.ogg', 5, 0)
+				if (15 to 29)
+					playsound(src, 'sound/items/timer.ogg', 30, 0)
+				if (0 to 14)
+					update_icon()
+					quiet  = !quiet
+					if(!quiet)
+						return
+					else
+						playsound(src, 'sound/machines/engine_alert2.ogg', 100, 0)
```

Comment:
The logic isn't sound here 
```
if(quiet)
    playsound ...
```
??

Either way, cut these 4 lines down to 2 by removing the return and inverting the logic